### PR TITLE
Add Example Python CDK Stack and Consolidate TypeScript CDK Stacks

### PR DESCRIPTION
**Pull request by duncanpharvey**
### What does this PR do?

* Adds example Python CDK stack with Node, Python, and Go Lambda functions
* Consolidates TypeScript CDK stacks to a single stack with the same Node, Python, and Go Lambda functions

### Motivation

https://datadoghq.atlassian.net/browse/SVLS-3956

### Testing Guidelines

Detailed in `README.md` of the example stacks.

### Types of Changes

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: examples/ts-stack-al2-arm-go-handler/README.md
Comment (duncanista) [inline examples/ts-stack-al2-arm-go-handler/README.md:1]: So no more docs for both stacks?

File: examples/ts-stack-js-python-handlers/README.md
Comment (duncanista) [inline examples/ts-stack-js-python-handlers/README.md:1]: This docs should be updated and re-added, I don't see them anymore on the new stacks.

**Review by duncanista — CHANGES_REQUESTED** [review]
Let's add the documentation back, I think it's very important. I imagine they need to be updated. Let's just make sure users can see those docs if they want to use it either in Python or Typescript.

File: examples/ts-stack-js-python-handlers/README.md
Comment (duncanpharvey) [reply examples/ts-stack-js-python-handlers/README.md:1]: Docs are in progress! PR isn't quite ready for review yet

File: examples/ts-stack-al2-arm-go-handler/README.md
Comment (duncanpharvey) [reply examples/ts-stack-al2-arm-go-handler/README.md:1]: Added READMEs for both example stacks!

File: examples/ts-stack-js-python-handlers/README.md
Comment (duncanpharvey) [reply examples/ts-stack-js-python-handlers/README.md:1]: Added READMEs for both example stacks!

File: examples/python-stack/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+Use this example Python stack to try out the [datadog-cdk-constructs](https://github.com/DataDog/datadog-cdk-constructs) v2 library. It contains Node, Python and Go Lambda functions.
+
+## Getting Started
+
+1. Get a Datadog API key to be used to send monitoring data ([Datadog API keys documentation](https://docs.datadoghq.com/account_management/api-app-keys/#add-an-api-key-or-client-token)).
+1. Run `export DD_API_KEY=DATADOG_API_KEY_HERE` to set the Datadog API key in your shell session.
+1. Run `virtualenv env` to create a virtual environment.
+1. Run `source env/bin/activate` to activate the virtual environment.
+1. Run `pip install -r requirements` install dependencies.
```
Comment (hestonhoffman) [inline examples/python-stack/README.md:11]:
```suggestion
1. Run `pip install -r requirements` to install dependencies.
```

File: examples/python-stack/README.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+1. Update the layer versions in `lib/cdk-python-stack.ts` with the latest releases:
+   - Datadog Lambda Extension: https://github.com/DataDog/datadog-lambda-extension/releases
+   - Python Layer: https://github.com/DataDog/datadog-lambda-python/releases
+   - Node Layer: https://github.com/DataDog/datadog-lambda-js/releases
+1. Run `cdk synth` to synthesize the CloudFormation template.
+1. Run `cdk diff` to see the resource and permission changes that are made.
+1. Run `cdk deploy` to deploy the stack to AWS
```
Comment (hestonhoffman) [inline examples/python-stack/README.md:18]:
```suggestion
1. Run `cdk deploy` to deploy the stack to AWS.
```

File: examples/python-stack/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+1. Invoke your Lambda functions and look for them in Datadog Serverless Monitoring
```
Comment (hestonhoffman) [inline examples/python-stack/README.md:19]:
```suggestion
1. Invoke your Lambda functions and look for them in Datadog Serverless Monitoring.
```

File: examples/typescript-stack/README.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Datadog CDK TypeScript Example
+
+Use this example TypeScript stack to try out the [datadog-cdk-constructs](https://github.com/DataDog/datadog-cdk-constructs) v2 library. It contains Node, Python and Go Lambda functions.
+
+## Getting Started
+
+1. Get a Datadog API key to be used to send monitoring data ([Datadog API keys documentation](https://docs.datadoghq.com/account_management/api-app-keys/#add-an-api-key-or-client-token)).
+1. Run `export DD_API_KEY=DATADOG_API_KEY_HERE` to set the Datadog API key in your shell session.
+1. Run `yarn` install dependencies.
```
Comment (hestonhoffman) [inline examples/typescript-stack/README.md:9]:
```suggestion
1. Run `yarn` to install dependencies.
```

File: examples/typescript-stack/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+1. Update the layer versions in `lib/cdk-typescript-stack.ts` with the latest releases:
+   - Datadog Lambda Extension: https://github.com/DataDog/datadog-lambda-extension/releases
+   - Python Layer: https://github.com/DataDog/datadog-lambda-python/releases
+   - Node Layer: https://github.com/DataDog/datadog-lambda-js/releases
+1. Run `cdk synth` to synthesize the CloudFormation template.
+1. Run `cdk diff` to see the resource and permission changes that are made.
+1. Run `cdk deploy` to deploy the stack to AWS
```
Comment (hestonhoffman) [inline examples/typescript-stack/README.md:16]:
```suggestion
1. Run `cdk deploy` to deploy the stack to AWS.
```

File: examples/typescript-stack/README.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+1. Invoke your Lambda functions and look for them in Datadog Serverless Monitoring
```
Comment (hestonhoffman) [inline examples/typescript-stack/README.md:17]:
```suggestion
1. Invoke your Lambda functions and look for them in Datadog Serverless Monitoring.
```

**Review by hestonhoffman — APPROVED** [review]
Left some minor edits, but LGTM

**Comment by duncanpharvey** [comment]
> Left some minor edits, but LGTM

Thanks @hestonhoffman! I committed your suggested changes

File: examples/python-stack/README.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# Datadog CDK Python Example
+
+Use this example Python stack to try out the [datadog-cdk-constructs](https://github.com/DataDog/datadog-cdk-constructs) v2 library. It contains Node, Python and Go Lambda functions.
+
+## Getting Started
+
+1. Get a Datadog API key to be used to send monitoring data ([Datadog API keys documentation](https://docs.datadoghq.com/account_management/api-app-keys/#add-an-api-key-or-client-token)).
+1. Run `export DD_API_KEY=DATADOG_API_KEY_HERE` to set the Datadog API key in your shell session.
```
Comment (duncanista) [inline examples/python-stack/README.md:8]: nit
```suggestion
1. Run `export DD_API_KEY=<DATADOG_API_KEY>` to set the Datadog API key in your environment.
```

File: examples/python-stack/README.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+1. Run `source env/bin/activate` to activate the virtual environment.
+1. Run `pip install -r requirements` to install dependencies.
+1. Update the layer versions in `lib/cdk-python-stack.ts` with the latest releases:
+   - Datadog Lambda Extension: https://github.com/DataDog/datadog-lambda-extension/releases
+   - Python Layer: https://github.com/DataDog/datadog-lambda-python/releases
+   - Node Layer: https://github.com/DataDog/datadog-lambda-js/releases
+1. Run `cdk synth` to synthesize the CloudFormation template.
+1. Run `cdk diff` to see the resource and permission changes that are made.
+1. Run `cdk deploy` to deploy the stack to AWS.
+1. Invoke your Lambda functions and look for them in Datadog Serverless Monitoring.
```
Comment (duncanista) [inline examples/python-stack/README.md:19]: Maybe we could add an image to the docs of what to expect? And maybe a link to the CDK configuration again?

File: examples/typescript-stack/README.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
+# Datadog CDK TypeScript Example
+
+Use this example TypeScript stack to try out the [datadog-cdk-constructs](https://github.com/DataDog/datadog-cdk-constructs) v2 library. It contains Node, Python and Go Lambda functions.
+
+## Getting Started
+
+1. Get a Datadog API key to be used to send monitoring data ([Datadog API keys documentation](https://docs.datadoghq.com/account_management/api-app-keys/#add-an-api-key-or-client-token)).
+1. Run `export DD_API_KEY=DATADOG_API_KEY_HERE` to set the Datadog API key in your shell session.
```
Comment (duncanista) [inline examples/typescript-stack/README.md:8]: Same as above.

File: examples/python-stack/README.md (hunk 7 above)
Comment (duncanpharvey) [reply examples/python-stack/README.md]: Updated to use `<DATADOG_API_KEY>`

File: examples/typescript-stack/README.md (hunk 9 above)
Comment (duncanpharvey) [reply examples/typescript-stack/README.md]: Updated to use `<DATADOG_API_KEY>`

File: examples/python-stack/README.md (hunk 8 above)
Comment (duncanpharvey) [reply examples/python-stack/README.md]: Added screenshots of a Lambda function trace in each readme, links directly to the Serverless page in Datadog for Lambda functions, and a permalink to the CDK configuration in each stack

**Pull request merged**